Documentation/Contributing.md

```diff
@@ -0,0 +1,348 @@
+```
+
+### Linux
+
+```bash
+$> ./swift/utils/build-script --preset=buildbot_swiftpm_linux_platform,tools=RA,stdlib=RA
+```
+
+Once the build is complete, you should be able to run the swiftpm binaries from the build folder.
+
+### Example
+
+```bash
+$> cd /tmp && mkdir hello && cd hello
+$> /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-package init
+$> /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-build
+```
+
+8. Test the Swift Package Manager.
+
+```bash
+$> Utilities/bootstrap test
+```
+
+## Getting Started
+1. Pull the SwiftPM repository:
+```bash
+$> git clone https://github.com/apple/swift-package-manager.git
+```
+2. Run your first build:
+```bash
+$> cd swift-package-manager
+$> swift build
+```
+Make sure the build did not fail.  
+
+A `.build/` folder will be generated and it should have inside a similar structure (including build binaries):
+```bash
+artifacts/
+checkouts/
+debug/
+repositories
+x86_64-apple-macosx
+debug.yaml
+manifest.db
+workspace-state.json 
+```
+Binaries (in the example above) are in `x86_64-apple-macosx/`.  
+If you need to build the generated binaries, run `swift-build` in inside `.build/`:
+```bash
+./.build/x86_64-apple-macosx/debug/swift-build
+```
+
+## Running Tests
+### Run all tests.
+```bash
+$> swift test --parallel
+```
+### Run a single test.
+```bash
+$> swift test --filter PackageGraphTests.DependencyResolverTests/testBasics
+```
+### Run tests for the test targets BuildTests and WorkspaceTests, but skip some test cases.
+```bash
+$> swift test --filter BuildTests --skip BuildPlanTests --filter WorkspaceTests --skip InitTests
+```
+Note: PackageDescription v4 is not available when developing using this method.
+
+This method can also used be used for performance testing. Use the following command run SwiftPM's performance tests:
+
+```bash
+$> export TSC_ENABLE_PERF_TESTS=1
+$> swift test -c release --filter PerformanceTests
+```
+
+For example, testing basic functionality using the newly created binaries 
+
+```bash
+$> cd /tmp && mkdir hello && cd hello
+$> /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-package init
+$> /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-build
+```
+
+## Testing on Linux with Docker
+For contributors on macOS who need to test on Linux, install Docker and use the
+following commands:
+
+```bash
+$> Utilities/Docker/docker-utils build # will build an image with the latest Swift snapshot
+$> Utilities/Docker/docker-utils bootstrap # will bootstrap SwiftPM on the Linux container
+$> Utilities/Docker/docker-utils run bash # to run an interactive Bash shell in the container
+$> Utilities/Docker/docker-utils swift-build # to run swift-build in the container
+$> Utilities/Docker/docker-utils swift-test # to run swift-test in the container
+$> Utilities/Docker/docker-utils swift-run # to run swift-run in the container
+```
+
+## Creating Pull Requests
```

Comment:
* please add a .GITHUB PR template to codify these expectations, see https://github.com/apple/swift-metrics/tree/main/.github

* please add a .GITHUB issue template codify these expectations, see https://github.com/apple/swift-metrics/tree/main/.github

* Add note on code style and how to use swift-format before submitting PRs (we just merged https://github.com/apple/swift-package-manager/pull/3012)

* Add comment on patch size, number of commits per PR, etc